core/bug_api.php

```diff
@@ -1895,52 +1979,69 @@ function bug_resolve( $p_bug_id, $p_resolution, $p_fixed_in_version = '', $p_bug
 	$c_resolution = (int)$p_resolution;
 	$p_bugnote_text = trim( $p_bugnote_text );
 
-	# Add bugnote if supplied
-	# Moved bugnote_add before bug_set_field calls in case time_tracking_no_note is off.
-	# Error condition stopped execution but status had already been changed
-	$t_bugnote_id = bugnote_add( $p_bug_id, $p_bugnote_text, $p_time_tracking, $p_bugnote_private, 0, '', null, false );
-	bugnote_process_mentions( $p_bug_id, $t_bugnote_id, $p_bugnote_text );
+	# @TODO
+	# If time tracking is enabled, this code is executed outside of BugData validation.
```

Comment:
I don't recall the reason why we do this special behavior for time tracking notes?  Do you know why?

Btw, @ TODOs are picked up by some IDEs like PhpStorm, but they show the first line next to them, so it is generally good practice to try the have the todo on the same line and having enough info on first line to provide an idea of what needs to be done.  There are multiple occurrences of such usage of @ TODOs.